developer/src/kmc-convert/src/keylayout-to-kmn/keylayout-to-kmn-converter.ts

```diff
@@ -1,27 +1,990 @@
+        add_modifier = "LALT ";
+      }
+      else if ((modifier_state[i].toUpperCase() === "RIGHTOPTION") || (modifier_state[i].toUpperCase() === "ROPTION")) {
+        add_modifier = "RALT ";
+      }
+      else if ((modifier_state[i].toUpperCase() === 'ANYOPTION') || (modifier_state[i].toUpperCase() === 'OPTION')) {
+        add_modifier = "RALT ";
+      }
+
+      else {
+        add_modifier = String(modifier_state[i]) + " ";
+      }
+      kmn_modifier += kmn_ncaps + add_modifier;
+    }
+    // remove duplicate and empty entries and make sure NCAPS is at the beginning
+    const duplicate_modifier_array: string[] = kmn_modifier.split(" ").filter(item => item);
+
+    const unique_modifier: string[] = duplicate_modifier_array.filter(function (item, pos, self) {
+      return self.indexOf(item) === pos;
+    });
+
+    return unique_modifier.flat().toString().replace(/,/g, " ");
+  }
+
+  /**
+   * @brief  member function to check if CAPS is used throughout a keylayout file or not
+   * @param  keylayout_modifier the modifier string used in the .keylayout-file
+   * @return kmn_modifier the modifier string used in the .kmn-file
+   */
+  public checkIfCapsIsUsed(keylayout_modifier: string[][]): boolean {
+    return JSON.stringify(keylayout_modifier).toUpperCase().includes("CAPS");
+  }
+
+  /**
+  * @brief  member function to check if a modifier can be used in keyman
+  * @param  keylayout_modifier the modifier string used in the .keylayout-file
+  * @return true if the modifier can be used in keyman; false if not
+  */
+  public isAcceptableKeymanModifier(keylayout_modifier: string): boolean {
+    if (keylayout_modifier === null)
+      return false;
+
+    let iskKeymanModifier: boolean = true;
+    const modifier_single: string[] = keylayout_modifier.split(" ");
+
+    for (let i = 0; i < modifier_single.length; i++) {
+      if (
+        (modifier_single[i].toUpperCase() === "NCAPS")
+        || (modifier_single[i].toUpperCase() === "CAPS")
+        || (modifier_single[i].toUpperCase() === "SHIFT")
+        || (modifier_single[i].toUpperCase() === "ALT")
+        || (modifier_single[i].toUpperCase() === "RALT")
+        || (modifier_single[i].toUpperCase() === "LALT")
+        || (modifier_single[i].toUpperCase() === "CTRL")
+        || (modifier_single[i].toUpperCase() === "LCTRL")
+        || (modifier_single[i].toUpperCase() === "RCTRL")
+        || (modifier_single[i].toUpperCase() === "")
+      ) {
+        iskKeymanModifier &&= true;
+      } else {
+        iskKeymanModifier &&= false;
+      }
+    }
+    return iskKeymanModifier;
+  }
+
+  /**
+   * @brief  member function to map Ukelele keycodes to Windows Keycodes
+   * @param  pos Ukelele (=mac) keycodes
+   * @return keycode on a Windows Keyboard
+   */
+  public map_UkeleleKC_To_VK(pos: number): string {
+
+    // ukelele KC  -->  VK_US
+    if (pos === 0x0A) return "K_BKQUOTE";         /* ^ */
```

Comment:
This long ladder of if/else could be an array -- first check bounds of pos:

```
  if(typeof pos == 'number' && pos >= 0 && pos <= 0x##) { 
    return UkeleleKC_to_VK[pos];
  }
  return "";
```

Then define Ukelele_to_VK as an array of strings:

```
const Ukelele_to_VK = [
  'K_A',
  'K_S',
...
];
```